dashboard/app/views/levels/show.html.haml

```diff
@@ -22,6 +22,9 @@
   :javascript
     var appOptions = #{app_options.to_json};
   = render partial: 'levels/unplug', locals: {app: @game.app}
+- elsif @level.is_a?(Applab) && app_options['readonlyWorkspace'] && !app_options[:channel]
+  -# A teacher is trying to view a student's progress on a level they have not started.
+  = render partial: 'levels/not_started'
```

Comment:
Nit: I think you can do `= render text: i18n.t(:not_started), layout: true` to avoid the partial.